ibis/expr/types/temporal.py

```diff
@@ -35,8 +270,50 @@ class TimeColumn(TemporalColumn, TimeValue):
 
 
 @public
-class DateValue(TemporalValue):
-    pass  # noqa: E701,E302
+class DateValue(DateComponentValue):
+    def truncate(self, unit: Literal["Y", "Q", "M", "W", "D"]) -> DateValue:
+        """Truncate date expression to units of `unit`.
+
+        Parameters
+        ----------
+        unit
+            Unit to truncate `arg` to
+
+        Returns
+        -------
+        DateValue
+            Truncated date value expression
+        """
+        import ibis.expr.operations as ops
+
+        return ops.DateTruncate(self, unit).to_expr()
+
+    def __add__(self, other: IntervalValue) -> DateValue:
+        """Add an interval to a date."""
+        import ibis.expr.operations as ops
+
+        return ops.DateAdd(self, other).to_expr()
+
+    add = radd = __radd__ = __add__
+
+    def __sub__(
+        self,
+        other: DateValue | IntervalValue,
+    ) -> IntervalValue | DateValue:
+        """Subtract a date or an interval from a date."""
+        import ibis.expr.operations as ops
+        import ibis.expr.rules as rlz
+
+        other = rlz.one_of([rlz.date, rlz.interval], other)
```

Comment:
Isn't this handled by the operation?

Reply:
I see now, we need to coerce in order to decide the type of `other`. 

Why are other `__sub__` implementations use `right = rlz.any(other)` instead of more specific rules?

Reply:
I think these aren't correct, fixing up.

Reply:
These are fixed now.